IntegrationTests/UI.IntegrationTests/CommandsDialogs/FormBrowseTests.cs

```diff
@@ -63,14 +63,8 @@ public void OneTimeTearDown()
         [SetUp]
         public void SetUp()
         {
-            if (_referenceRepository is null)
-            {
-                _referenceRepository = new ReferenceRepository();
-            }
-            else
-            {
-                _referenceRepository.Reset();
-            }
+            // we need to completely reset each time
+            _referenceRepository = new ReferenceRepository();
```

Comment:
This slows downs tests, but only some tenths of a second locally, so it is OK.

Reply:
The tests on that class needs clean repositories, and this is the way to reset it that I found on the code (FormBrowse.LeftPanelTests.cs:72 and FormBrowse.LeftPanelTests.Remotes.cs:61).  Is there a better way?

Reply:
`_referenceRepository` must be disposed of to clean up after itself.

> The tests on that class needs clean repositories

This is precisely what `_referenceRepository.Reset()` is doing, it performs `reset --hard` on the very first commit.
https://github.com/gitextensions/gitextensions/blob/4fd3843999ab4b25b090ce4eea364c15e180ece2/UnitTests/CommonTestUtils/ReferenceRepository.cs#L92-L112

Reply:
In this case we must dispose `_referenceRepository` in `TearDown`, which is run for every test. https://docs.nunit.org/articles/nunit/technical-notes/usage/SetUp-and-TearDown.html